neutrinordp/xrdp-neutrinordp.c

```diff
@@ -496,6 +503,10 @@ lxrdp_set_param(struct mod *mod, const char *name, const char *value)
     {
         settings->desktop_resize = g_text2bool(value);
     }
+    else if (g_strcmp(name, "pampassword") == 0)
+    {
+        LOG(LOG_LEVEL_WARNING, "lxrdp_set_param: unknown name [%s] value [******]", name);
```

Comment:
Do we need to log this? It's a valid parameter in some contexts. I'd got for logging nothing at all for `pamusername`, `pampassword` and `pamsessionmng`

Reply:
Indeed. What about doing nothing?

```
    else if (g_strcmp(name, "pampassword") == 0)
    {
    }
```

Reply:
Also shouldn't `pampassword` and `pamsessionmng` reach `lxrdp_set_param: unknown name` at the bottom?

Reply:
Well  `pamusername`, `pampassword` and `pamsessionmng` are all valid settings for this module even if they're not used by it, so I'd vote for not logging them at all:-

```c
    else if (g_strcmp(name, "pamusername") == 0 ||
             g_strcmp(name, "pampassword") == 0 ||
             g_strcmp(name, "pamsessionmng") == 0)
    {
        /* Valid (but unused) parameters not logged */
    }
    else
    {
        LOG(LOG_LEVEL_WARNING, "lxrdp_set_param: unknown name [%s] value [%s]", name, value);
    }
   . . .
```

The counter-argument is that keeping the above list maintained is not going to happen, so maybe we should just stick with omitting the security-sensitive values.

Reply:
This is the suggested change by the reporter @TOMATO-ONE

```diff
--- xrdp/neutrinordp/xrdp-neutrinordp.c.orig
+++ xrdp/neutrinordp/xrdp-neutrinordp.c
@@ -502,6 +502,10 @@
     {
         settings->nla_security = g_text2bool(value);
     }
+    else if (g_strcmp(name, "pampassword") == 0)
+    {
+       LLOGLN(0, ("lxrdp_set_param: unknown name [%s] value [%s]", name, "*****" ));
+    }
     else
     {
         LLOGLN(0, ("lxrdp_set_param: unknown name [%s] value [%s]", name, value));
```

And this is what I'm thinking of:
```diff
+    else if (g_strcmp(name, "pampassword") == 0 ||
+             g_strcmp(name, "pamusername") == 0 ||
+             g_strcmp(name, "pammsessionmng") == 0)
+    {
+        /*
+         * Doing nothing on purpose
+         *
+         * `pampassword will contain raw password
+         * `pamusername` and `pamsessionmng` are also valid parameters in some contexts
+         *  so not to log "unknown name"
+         *
+         */
+    }
```

Reply:
I agree with both of  @matt335672 points. I also prefer your change @metalefty rather than the reporter's change.

Reply:
Thank you.
My patch was tentative.
I also think your patch is better.